ducktape/tests/test.py

```diff
@@ -151,16 +150,19 @@ def __init__(self, **kwargs):
         self.injected_args = kwargs.get("injected_args", None)
         self.ignore = kwargs.get("ignore", False)
 
-        self.services = kwargs.get("services", ServiceRegistry())
+        self.services = ServiceRegistry()
 
         # dict for toggling service log collection on/off
-        self.log_collect = kwargs.get("log_collect", {})
+        self.log_collect = {}
 
     def __repr__(self):
         return "<module=%s, cls=%s, function=%s, injected_args=%s>" % \
                (self.module, self.cls_name, self.function_name, str(self.injected_args))
 
-    def copy(self, **kwargs):
+    def copy_constructor(self, **kwargs):
```

Comment:
Yes this is a good point. My concern with `copy` is that it suggests it actually copies the full original object. That's not quite true since the newly constructed object does not include a copy of the other object's `ServiceRegistry` object for example.

 `copy_constructor` is maybe not that much better of a name in this regard...